homeassistant/components/withings/binary_sensor.py

```diff
@@ -45,21 +21,36 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the sensor config entry."""
-    coordinator: WithingsDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-
-    entities = [
-        WithingsBinarySensor(coordinator, attribute) for attribute in BINARY_SENSORS
-    ]
+    user_id = str(entry.unique_id)
 
-    async_add_entities(entities)
+    async_add_entities([WithingSleepBinarySensor(user_id)])
 
 
-class WithingsBinarySensor(WithingsEntity, BinarySensorEntity):
+# Scheduled for removal in 2024.4
+class WithingSleepBinarySensor(BinarySensorEntity):
     """Implementation of a Withings sensor."""
 
-    entity_description: WithingsBinarySensorEntityDescription
-
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if the binary sensor is on."""
-        return self.coordinator.in_bed
+    _attr_has_entity_name = True
+    _attr_translation_key = "in_bed"
+    _attr_icon = "mdi:bed"
+    _attr_device_class = BinarySensorDeviceClass.OCCUPANCY
+
+    def __init__(self, user_id: str) -> None:
+        """Initialize sleep sensor."""
+        self._user_id = user_id
+        self._attr_unique_id = f"withings_{user_id}_{Measurement.IN_BED.value}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, user_id)},
+            manufacturer="Withings",
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Listen to events after being added."""
+        await super().async_added_to_hass()
+        self.hass.bus.async_listen(f"withings_{self._user_id}_sleep", self.handle_event)
```

Comment:
We should use the dispatcher here instead.

The recorder, and other integrations listen to the event bus have process every event (and save it in the db in the case of the record).